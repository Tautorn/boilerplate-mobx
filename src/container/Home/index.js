import React from 'react'
import { observer } from 'mobx-react'
import map from 'lodash/map'
import HomeStore from './store'

class Home extends React.Component {

  componentWillMount() {
    this.store = new HomeStore()
  }

  renderList({id, status, created, updated}, index) {
    return (
      <tr key={index}>
        <td>{id}</td>
        <td>{status}</td>
        <td>{created}</td>
        <td>{updated}</td>
      </tr>
    )
  }

  render() {

    const { handleChange, msg, list } = this.store

    return (
      <div>
        <span>Hello</span>
        <p>{msg}</p>
        <input type="text" onChange={(e) => handleChange(e.target.value)} />
        <table>
          <thead>
            <tr>
              <th>My List</th>
            </tr>
          </thead>>
          <tbody>
            {list && map(map, (val, index) => this.renderList(val, index))}
          </tbody>
        </table>
      </div>
    )
  }
}

export default observer(Home)
