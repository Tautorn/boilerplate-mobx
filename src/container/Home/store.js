import { observable, action } from 'mobx'
import HomeProvider from 'core/provider/home'

class HomeStore {

  @observable msg = 'Test'
  @observable list = []

  constructor() {
    this.handleChange = this.handleChange.bind(this)
  }

  @action handleChange(value) {
    this.msg = value
  }

  loadStore() {
    HomeProvider.search()
      .then(({ res }) => this.list = res.data )
      .catch(( err ) => console.log('exception', err))
  }



}

export default HomeStore
