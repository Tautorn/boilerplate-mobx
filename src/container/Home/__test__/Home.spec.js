import React from 'react'
import { shallow } from 'enzyme'
import Home from '../'

describe('Home', () => {

  it('should render login ', () => {
    const wrapper = shallow(<Home />)

    expect(wrapper).toMatchSnapshot()
  })

})
