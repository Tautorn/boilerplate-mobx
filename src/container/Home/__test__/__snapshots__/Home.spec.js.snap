// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render login  1`] = `
<div>
  <span>
    Hello
  </span>
  <p>
    Test
  </p>
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
